src/Runner.Listener/MessageListener.cs

```diff
@@ -120,8 +163,20 @@ public async Task<Boolean> CreateSessionAsync(CancellationToken token)
 
                     if (!IsSessionCreationExceptionRetriable(ex))
                     {
-                        _term.WriteError($"Failed to create session. {ex.Message}");
-                        return false;
+                        if (_useV2Credentials && _v1CredentialsExists)
+                        {
+                            // v2 credentials might cause lose permission during permission check,
+                            // we will force to use v1 credential and try again
+                            _useV2Credentials = false;
+                            _authorizationUrlRollbackReattemptDelayBackgroundTask = HostContext.Delay(TimeSpan.FromDays(2), token); // retry v2 creds in 2 days.
```

Comment:
what is the motivation for such a long delay?

Reply:
During the initial rollout, I am trying to avoid wasting time trying Token service if there is a service bug and we need to do a binary deployment to fix the problem in Token service.
I can short the back-off to 12-24 hours